ksqldb-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -160,13 +170,12 @@ public void onStop(final Consumer<Boolean> onStop) {
     this.onStop = onStop;
   }
 
-  protected void uncaughtHandler(final Thread t, final Throwable e) {
+  protected StreamsUncaughtExceptionHandler.StreamThreadExceptionResponse uncaughtHandler(final Throwable e) {
     QueryError.Type errorType = Type.UNKNOWN;
     try {
       errorType = errorClassifier.classify(e);
     } catch (final Exception classificationException) {
       LOG.error("Error classifying unhandled exception", classificationException);
-      throw classificationException;
```

Comment:
We don't want to throw errors from the handler this will cause streams the silently kill threads. Either we restart like normal or we kill the client/application. This is set to retry here.